src/core/thread/mesh_forwarder.cpp

```diff
@@ -909,6 +911,19 @@ void MeshForwarder::HandleSentFrame(Mac::TxFrame &aFrame, otError aError)
         mMessageNextOffset = mSendMessage->GetLength();
 #endif
     }
+#if OPENTHREAD_THREAD_VERSION >= OPENTHREAD_THREAD_VERSION_1_2
+    else if (aAckFrame != NULL)
```

Comment:
I think it's necessary to check here that the parent is Thread Version 1.2 or greater. There is no guarantee that all Thread 1.1 devices are kind about setting their FP bit in response to data messages. Also resetting the keepalive for all data messages to 1.1 parent would break backwards compatibility.